File: Python/program/sivitas.py
# Saya Muhamad Firdaus [2101995] mengerjakan
# soal Latihan 4 dalam mata kuliah DPBO
# untuk keberkahanNya maka saya tidak melakukan kecurangan 
# seperti yang telah dispesifikasikan. Aamiin.

from human import Human

class SivitasAkademik(Human):
    # Constructor
    def __init__(self, NIK, nama, jenis_kelamin, asal_universitas, email_edu):
        Human.__init__(self, NIK, nama, jenis_kelamin)
        self.asal_universitas = asal_universitas
        self.email_edu = email_edu

    # Setter dan Getter
    def setUniversitas(self, asal_universitas):
        self.asal_universitas = asal_universitas

    def getUniversitas(self):
        return self.asal_universitas

    def setEmail_edu(self, email_edu):
        self.email_edu = email_edu

    def getEmail_edu(self):
        return self.email_edu

File: cpp/program/sivitas.cpp
// Saya Muhamad Firdaus [2101995] mengerjakan
// soal Latihan 4 dalam mata kuliah DPBO
// untuk keberkahanNya maka saya tidak melakukan kecurangan 
// seperti yang telah dispesifikasikan. Aamiin.

#include "header.hh"

// Constructor 
SivitasAkademik::SivitasAkademik(){
    this->asal_universitas = "";
    this->email_edu = "";
}

// Constructor
SivitasAkademik::SivitasAkademik(string NIK, string nama, string jenis_kelamin, string asal_universitas, string email_edu) : 
Human(NIK, nama, jenis_kelamin){
this->asal_universitas = asal_universitas;
this->email_edu = email_edu;
}
 
// Setter dan Getter
void SivitasAkademik::setUniversitas(string asal_universitas){
    this->asal_universitas = asal_universitas;
}
    
string SivitasAkademik::getUniversitas(){
    return asal_universitas;
}

void SivitasAkademik::setEmail_edu(string email_edu){
    this->email_edu = email_edu;
}

string SivitasAkademik::getEmail_edu(){
    return email_edu;
}

// Destructor
SivitasAkademik::~SivitasAkademik(){

}
